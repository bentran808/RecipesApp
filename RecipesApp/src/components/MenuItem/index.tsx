import React from 'react';
import isEqual from 'react-fast-compare';
import { Image, ImageSourcePropType, Text, TouchableHighlight, View } from 'react-native';
import styles from './styles';

type Props = {
  title: string;
  onPress: () => void;
  source: ImageSourcePropType;
};

const MenuItem = ({ title, onPress, source }: Props) => {
  return (
    <TouchableHighlight
      onPress={onPress}
      style={styles.btnClickContain}
      underlayColor="rgba(128, 128, 128, 0.1)"
    >
      <View style={styles.btnContainer}>
        <Image source={source} style={styles.btnIcon} />
        <Text style={styles.btnText}>{title}</Text>
      </View>
    </TouchableHighlight>
  );
};

export default React.memo(MenuItem, isEqual);
