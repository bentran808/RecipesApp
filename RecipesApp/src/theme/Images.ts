export const BackArrow = require('assets/icons/backArrow.png');
export const MenuIcon = require('assets/icons/menu.png');
export const HomeIcon = require('assets/icons/home.png');
export const CategoryIcon = require('assets/icons/category.png');
export const SearchIcon = require('assets/icons/search.png');
