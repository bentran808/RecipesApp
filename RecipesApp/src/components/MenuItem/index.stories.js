import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { View } from 'react-native';
import CenterView from 'storybook/stories/CenterView';
import { HomeIcon } from 'theme';
import MenuItem from '.';

storiesOf('Menu Item', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => (
    <MenuItem title="Hello Menu Item" source={HomeIcon} onPress={action('clicked-item')} />
  ));
