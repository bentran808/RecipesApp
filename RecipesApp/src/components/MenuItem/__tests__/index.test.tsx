import MenuItem from 'components/MenuItem';
import renderer from 'react-test-renderer';
import { HomeIcon } from 'theme';

describe('Menu Item Component', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<MenuItem title="Home" onPress={jest.fn()} source={HomeIcon} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
