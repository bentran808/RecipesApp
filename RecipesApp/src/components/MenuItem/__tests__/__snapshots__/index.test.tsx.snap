// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "marginVertical": 5,
      "padding": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 10,
          "marginTop": 2,
        }
      }
    >
      Home
    </Text>
  </View>
</View>
`;
